File: API/Features/Reservations/Reservations/ViewModels/List/ReservationListVM.cs
using System;
using API.Infrastructure.Classes;

namespace API.Features.Reservations.Reservations {

    public class ReservationListVM {

        public Guid ReservationId { get; set; }

        public string Date { get; set; }
        public string RefNo { get; set; }
        public string TicketNo { get; set; }
        public int Adults { get; set; }
        public int Kids { get; set; }
        public int Free { get; set; }
        public int TotalPax { get; set; }

        public SimpleEntity Customer { get; set; }
        public ReservationListCoachRouteVM CoachRoute { get; set; }
        public ReservationListDestinationVM Destination { get; set; }
        public ReservationListDriverVM Driver { get; set; }
        public ReservationListPickupPointVM PickupPoint { get; set; }
        public ReservationListPortVM Port { get; set; }
        public SimpleEntity Ship { get; set; }

        public int PassengerCount { get; set; }
        public int PassengerDifference { get; set; }

    }

}

File: FrontEnd/src/app/features/reservations/reservations/classes/view-models/list/reservation-list-vm.ts
import { ReservationListCoachRouteVM } from './reservation-list-coachRoute-vm'
import { ReservationListDestinationVM } from './reservation-list-destination-vm'
import { ReservationListDriverVM } from './reservation-list-driver-vm'
import { ReservationListPickupPointVM } from './reservation-list-pickupPoint-vm'
import { ReservationListPortVM } from './reservation-list-port-vm'
import { ReservationListShipVM } from './reservation-list-ship-vm'
import { SimpleEntity } from './../../../../../../shared/classes/simple-entity'

export interface ReservationListVM {

    reservationId: string
    date: string
    adults: number
    kids: number
    free: number
    totalPax: number
    ticketNo: string
    coachRoute: ReservationListCoachRouteVM
    customer: SimpleEntity
    destination: ReservationListDestinationVM
    driver: ReservationListDriverVM
    pickupPoint: ReservationListPickupPointVM
    port: ReservationListPortVM
    ship: ReservationListShipVM
    user: string

}
